### PR TITLE
AuditAPI CodeQualityEvaluator - Hot House Revisions Implementation

### PR DESCRIPTION
**Pull request by kumarvarun1252**
Implemented new audit statuses according to hothouse revisions for SCA - CodeQuality

sca audit status - hot house revision

### CODE REVIEW

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java [hunk 1]
```diff
@@ -100,18 +108,50 @@ private CodeQualityAuditResponse getStaticAnalysisResponse(CollectorItem collect
...
+                        } else if(condition.get("metric").toString().equalsIgnoreCase(CodeQualityMetricType.UNIT_TEST.getType()) &&
+                                condition.containsKey("error")) {
+                            codeQualityAuditResponse.addAuditStatus(CodeQualityAuditStatus.CODE_QUALITY_THRESHOLD_UNIT_TEST_FOUND);
+                        } else if(condition.get("metric").toString().equalsIgnoreCase(CodeQualityMetricType.NEW_COVERAGE.getType()) &&
+                                condition.containsKey("error")) {
+                            codeQualityAuditResponse.addAuditStatus(CodeQualityAuditStatus.CODE_QUALITY_THRESHOLD_CODE_COVERAGE_FOUND);
+                        }
+                    }
+                } catch (ParseException e) {
+                    e.printStackTrace();
```
Comment (rvema) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java]: please log the exception not print the stacktrace

File: api-audit/src/main/java/com/capitalone/dashboard/response/AuditReviewResponse.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package com.capitalone.dashboard.response;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.SortedSet;
```
Comment (rvema) [inline api-audit/src/main/java/com/capitalone/dashboard/response/AuditReviewResponse.java]: Please refrain from changing the signature of the response.

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java [hunk 3]
```diff
@@ -96,22 +103,52 @@ private CodeQualityAuditResponse getStaticAnalysisResponse(CollectorItem collect
...
+                    Iterator itr = conditions.iterator();
+
+                    // Iterate through the quality_gate conditions
+                    while(itr.hasNext()) {
+                        Map condition = ((Map) itr.next());
+
+                        // Set audit statuses for Thresholds found if they are defined in quality_gate_details metric
+                        if(condition.containsKey("error")) {
+                            if(condition.get("metric").toString().equalsIgnoreCase(CodeQualityMetricType.BLOCKER_VIOLATIONS.getType())) {
+                                codeQualityAuditResponse.addAuditStatus(CodeQualityAuditStatus.CODE_QUALITY_THRESHOLD_BLOCKER_FOUND);
```
Comment (rvema) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java]: Please use StringUtils.equalsIgnoreCase

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java [hunk 4]
```diff
@@ -123,15 +160,26 @@ private CodeQualityAuditResponse getStaticAnalysisResponse(CollectorItem collect
...
-                        e.printStackTrace();
+                        LOGGER.error("Error in CodeQualityEvaluator.getStaticAnalysisResponse() - Unable to parse quality_gate metrics - " + e.getMessage());
                     }
-
                 }
+            }
 
+            // Set audit statuses if the threshold is met based on the status field of code_quality metric
+            if (metric.getStatus().equals(CodeQualityMetricStatus.Ok)){
+                if (metric.getName().equalsIgnoreCase(CodeQualityMetricType.BLOCKER_VIOLATIONS.getType())) {
```
Comment (rvema) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java]: Please StringUtils

File: api-audit/src/test/resources/component/component.json [hunk 5]
```diff
@@ -72,12 +63,40 @@
           "instanceUrl": "https://scan.com"
         }
       }
+    ],
+    "Test" : [
+      {
+        "id" : "57f2a0193b55670a9e06d63a",
+        "description" : "ease-COAF-UI-ATDD-Functional-SLATE",
+        "niceName" : "COAF",
```
Comment (rvema) [inline api-audit/src/test/resources/component/component.json]: avoid using capital one specific terms in test data

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java [hunk 6]
```diff
@@ -96,22 +103,52 @@ private CodeQualityAuditResponse getStaticAnalysisResponse(CollectorItem collect
...
+                codeQualityAuditResponse.addAuditStatus(CodeQualityAuditStatus.CODE_QUALITY_GATES_FOUND);
+                try {
+                    JSONObject qualityGateDetails = (JSONObject) new JSONParser().parse(metric.getValue().toString());
+                    JSONArray conditions = (JSONArray) qualityGateDetails.get("conditions");
+                    Iterator itr = conditions.iterator();
+
+                    // Iterate through the quality_gate conditions
+                    while(itr.hasNext()) {
+                        Map condition = ((Map) itr.next());
+
```
Comment (rvema) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java]: use StringUtils

File: api-audit/src/test/resources/component/component.json [hunk 7]
```diff
@@ -72,12 +63,40 @@
...
+        "enabled" : true,
+        "errors" : [
+        ],
+        "pushed" : true,
+        "collectorId" : "56df29df7fab7c3ec3c432ea",
+        "lastUpdated" : 1513106044147,
+        "options" : {
+          "jobName" : "ease-COAF-UI-ATDD-Functional-SLATE",
```
Comment (rvema) [inline api-audit/src/test/resources/component/component.json]: please change the jobname

**Comment by kumarvarun1252** [comment]
@rvema I have addressed all you comments and made the changes as you suggested. Please review

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java [hunk 8]
```diff
@@ -33,6 +36,11 @@
 import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
+import java.util.Iterator;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.*;
```
Comment (rvema) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java]: Please import the classes you need and not a generic import

Comment (kumarvarun1252) [reply api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeQualityEvaluator.java]: done with this change

**Pull request merged**